src/script/lua_api/l_localplayer.cpp

```diff
@@ -0,0 +1,415 @@
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License along
+with this program; if not, write to the Free Software Foundation, Inc.,
+51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include "script/common/c_converter.h"
+#include "l_localplayer.h"
+#include "l_internal.h"
+
+LuaLocalPlayer::LuaLocalPlayer(LocalPlayer *m) {
+	m_localplayer = m;
+}
+
+void LuaLocalPlayer::create(lua_State *L, LocalPlayer *m) {
+	LuaLocalPlayer *o = new LuaLocalPlayer(m);
+	*(void **) (lua_newuserdata(L, sizeof(void *))) = o;
+	luaL_getmetatable(L, className);
+	lua_setmetatable(L, -2);
+
+	// Keep localplayer object stack id
+	int localplayer_object = lua_gettop(L);
+
+	lua_getglobal(L, "core");
+	lua_getfield(L, -1, "localplayer");
+	luaL_checktype(L, -1, LUA_TTABLE);
+	int uitable = lua_gettop(L);
+
+	lua_pushvalue(L, localplayer_object);
+	lua_setfield(L, uitable, "localplayer");
+}
+
+int LuaLocalPlayer::l_get_speed(lua_State *L) {
+	LocalPlayer *player = getobject(L, 1);
+
+
+	push_v3f(L, player->getSpeed());
+	return 1;
+}
+
+int LuaLocalPlayer::l_get_movement_speed_walk(lua_State *L) {
+	LocalPlayer *player = getobject(L, 1);
+
+	lua_pushnumber(L, player->movement_speed_walk);
+	return 1;
+}
+
+int LuaLocalPlayer::l_get_hp(lua_State *L) {
+	LocalPlayer *player = getobject(L, 1);
+
+	lua_pushinteger(L, player->hp);
+	return 1;
+}
+
+int LuaLocalPlayer::l_got_teleported(lua_State *L) {
+	LocalPlayer *player = getobject(L, 1);
+
+	lua_pushboolean(L, player->got_teleported);
+	return 1;
+}
+
+int LuaLocalPlayer::l_is_attached(lua_State *L) {
+	LocalPlayer *player = getobject(L, 1);
+
+	lua_pushboolean(L, player->isAttached);
+	return 1;
+}
+
+int LuaLocalPlayer::l_is_touching_ground(lua_State *L) {
+	LocalPlayer *player = getobject(L, 1);
+
+	lua_pushboolean(L, player->touching_ground);
+	return 1;
+}
+
+int LuaLocalPlayer::l_is_in_liquid(lua_State *L) {
+	LocalPlayer *player = getobject(L, 1);
+
+	lua_pushboolean(L, player->in_liquid);
+	return 1;
+}
+
+int LuaLocalPlayer::l_is_in_liquid_stable(lua_State *L) {
+	LocalPlayer *player = getobject(L, 1);
+
+	lua_pushboolean(L, player->in_liquid_stable);
+	return 1;
+}
+
+int LuaLocalPlayer::l_get_liquid_viscosity(lua_State *L) {
+	LocalPlayer *player = getobject(L, 1);
+
+	lua_pushinteger(L, player->liquid_viscosity);
+	return 1;
+}
+
+int LuaLocalPlayer::l_is_climbing(lua_State *L) {
+	LocalPlayer *player = getobject(L, 1);
+
+	lua_pushboolean(L, player->is_climbing);
+	return 1;
+}
+
+int LuaLocalPlayer::l_swimming_vertical(lua_State *L) {
+	LocalPlayer *player = getobject(L, 1);
+
+	lua_pushboolean(L, player->swimming_vertical);
+	return 1;
+}
+
+int LuaLocalPlayer::l_get_physics_override_speed(lua_State *L) {
```

Comment:
I agree that these parameters should be bundled together. It reduces the function count dramatically and easily allows adding more parameters to the returned table.